src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb

```diff
@@ -171,7 +170,7 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
                     ' If this is the "default" settings file, we add the "My" module as well...
                     '
                     If shouldGenerateMyStuff Then
-                        AddMyModule(CompileUnit, projectRootNamespace, wszDefaultNamespace)
+                        AddMyModule(CompileUnit, projectRootNamespace, wszDefaultNamespace.Replace(" ", "_"))
```

Comment:
I believe there are other places in Microsoft.VisualStudio.Editors and/or Microsoft.VisualStudio.AppDesigner where we need to munge a file name into a valid VB or C# identifier. It's probably worth spending a few minutes looking around to see how that is done, and whether it can be re-used here. If so, that would take care of other potentially problematic characters beyond the space.

That being said, I don't want the perfect to be the enemy of the good, so don't spend too long on that.

Reply:
It turns out there are already existing methods to generate valid identifiers. These are reused now for both class and namespace